src/util/merkleblock.rs

```diff
@@ -420,32 +421,40 @@ impl MerkleBlock {
     /// // Create a merkle block containing a single transaction
     /// let txid = Txid::from_hex(
     ///     "5a4ebf66822b0b2d56bd9dc64ece0bc38ee7844a23ff1d7320a88c5fdb2ad3e2").unwrap();
-    /// let match_txids = vec![txid].into_iter().collect();
-    /// let mb = MerkleBlock::from_block(&block, &match_txids);
+    /// let match_txids: Vec<Txid> = vec![txid].into_iter().collect();
+    /// let mb = MerkleBlock::from_block_with_predicate(&block, |t| match_txids.contains(t));
     ///
     /// // Authenticate and extract matched transaction ids
     /// let mut matches: Vec<Txid> = vec![];
     /// let mut index: Vec<u32> = vec![];
     /// assert!(mb.extract_matches(&mut matches, &mut index).is_ok());
     /// assert_eq!(txid, matches[0]);
     /// ```
-    pub fn from_block(block: &Block, match_txids: &HashSet<Txid>) -> Self {
+    pub fn from_block_with_predicate<F>(block: &Block, match_txids: F) -> Self
+        where F: Fn(&Txid) -> bool {
         let block_txids: Vec<_> = block.txdata.iter().map(Transaction::txid).collect();
-        Self::from_header_txids(&block.header, &block_txids, match_txids)
+        Self::from_header_txids_with_predicate(&block.header, &block_txids, match_txids)
     }
 
-    /// Create a MerkleBlock from the block's header and txids, that should contain proofs for match_txids.
+    /// Create a MerkleBlock from a block, that contains proofs for specific txids.
+    #[cfg(feature = "std")]
+    #[deprecated(since="0.26.2", note="use from_block_p")]
```

Comment:
This needs to follow the rename to use `predicate` instead of `p`.

Reply:
oops, fixed